src/pages/Messages.jsx

```diff
@@ -22,51 +20,29 @@ import { MessageButtonGroup, MessageFolder } from '../components/Messages';
  */
 const Messages = () => {
   localStorage.setItem('restorePath', '/messages');
-  const { podUrl } = useContext(SignedInUserContext);
-  const { session } = useSession();
+  const { updateMessageCountState } = useContext(MessageContext);
   const {
-    inboxList,
-    setInboxList,
-    outboxList,
-    setOutboxList,
-    loadMessages,
-    setLoadMessages,
-    updateMessageCountState
-  } = useContext(MessageContext);
+    data: inboxList,
+    refetch: refreshInbox,
+    isFetching: fetchingInbox
+  } = useMessageList('Inbox');
+  const {
+    data: outboxList,
+    refetch: refreshOutbox,
+    isFetching: fetchingOutbox
+  } = useMessageList('Outbox');
 
   // Handler function for refreshing PASS messages
-  const handleMessageRefresh = async (folderType) => {
-    setLoadMessages(true);
-    const messageList = folderType === 'Inbox' ? inboxList : outboxList;
-    const messagesInSolid = await getMessageTTL(session, folderType, messageList, podUrl);
-    messagesInSolid.sort((a, b) => b.uploadDate - a.uploadDate);
-    if (folderType === 'Inbox') {
-      setInboxList(messagesInSolid);
-    } else {
-      setOutboxList(messagesInSolid);
-    }
-    setLoadMessages(false);
+  const handleMessageRefresh = async () => {
```

Comment:
Probably want to create condition to trigger just one of the manual refetches rather than both. A simple if/else should do.